src/components/Select/Select.js

```diff
@@ -0,0 +1,147 @@
+  label: select;
+  appearance: none;
+  background-color: ${theme.colors.white};
+  border-width: 1px;
+  border-style: solid;
+  border-color: ${theme.colors.n300};
+  border-radius: ${theme.borderRadius.mega};
+  box-shadow: inset 0 1px 2px 0 rgba(102, 113, 123, 0.21);
+  color: ${theme.colors.n900};
+  padding: ${theme.spacings.byte} ${theme.spacings.tera} ${theme.spacings.byte}
+    ${theme.spacings.kilo};
+  position: relative;
+  ${textMega({ theme })};
+  z-index: 30;
+
+  &:focus,
+  &:hover,
+  &:active {
+    outline: none;
+  }
+
+  &:focus {
+    border: 1px solid ${theme.colors.b500};
+  }
+
+  &::after {
+  }
+
+  &:-moz-focusring {
+    color: transparent;
+    text-shadow: 0 0 0 #000;
+  }
+`;
+
+const disabledSelectStyles = ({ disabled }) =>
+  disabled &&
+  css`
+    label: select--disabled;
+    opacity: 0.4;
+    pointer-events: none;
+  `;
+
+const baseIconStyles = ({ theme }) => css`
+  label: select__icon;
+  fill: ${theme.colors.n700};
+  display: block;
+  z-index: 40;
+  pointer-events: none;
+`;
+
+const SelectElement = styled('select', { label: 'SelectElement' })(
+  baseSelectStyles,
+  disabledSelectStyles
+);
+const Icon = styled(DownTriangleIcon, { label: 'SelectIcon' })(baseIconStyles);
+
+/**
+ * A native select component.
+ */
+const Select = ({
+  options,
+  value,
+  placeholder,
+  selector,
+  disabled,
+  ...props
+}) => (
+  <IconInputWrapper
+    {...{ selector }}
+    iconPosition="right"
+    icon={({ className, disabledClassName }) => (
+      <Icon className={cx(className, { [disabledClassName]: disabled })} />
+    )}
+    input={({ className }) => (
+      <SelectElement {...{ ...props, className, disabled }}>
+        {!value && <option key={0}>{placeholder}</option>}
+        {options &&
+          options.map(option => (
+            <option key={option.value} {...option}>
+              {option.label}
+            </option>
+          ))}
+      </SelectElement>
+    )}
+  />
+);
+
+Select.propTypes = {
+  /**
+   * An ID rendered as data-selector attribute on the
+   * component. Used for tracking and e2e testing.
+   */
+  selector: PropTypes.string.isRequired,
+  /**
+   * onChange handler, called when the selection changes.
+   */
+  onChange: PropTypes.func.isRequired,
+  /**
+   * Name of the select form element.
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * Options to select from.
+   */
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+        .isRequired,
+      label: PropTypes.string,
+      disabled: PropTypes.bool
+    })
+  ).isRequired,
+  /**
+   * Styles the select as disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Currently selected value. Matches the "value" property of
+   * the options objects. If value is falsy, Select will render
+   * the "placeholder" prop as currently selected.
+   */
+  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```

Comment:
Is this a reasonable assumption? Can we only accept strings and numbers? For instance, in Angular you can have selected objects. Is this something we can support or want to support? 

I'm not sure why we actually need to define this - exactly the same as my comments on `{children}` over an options object, requiring `onChange` and even `placeholder` -- why put this into your abstraction and not allow this to be totally handled by the end user?

I imagine something like this, which is a much thinner/closer to the DOM:

```
<Select selectors={{ test: "shipping-address@signup' }}>
  <option value="">My default option</option>
  <option value="a" data-test="shipping-address__a@signup">A</option>
</Select>
```

LMK what you think about this. Of course, if we make a fancier select with images and autocomplete we'll have to think about all this stuff then - but no need to assume these things for our forms from the get go.

Reply:
Hm yeah. Agreed. Doesn't have to be this way. I guess I was trying to make something an easy abstraction forgetting that a) it might not be easy b) won't cover enough edge cases.